src/coreclr/tools/Common/Internal/Runtime/RuntimeConstants.cs

```diff
@@ -73,5 +73,8 @@ internal static class MethodFixupCellFlagsConstants
         public const int IsObjectiveCMessageSendMask = 0x8;
         public const int ObjectiveCMessageSendFunctionMask = 0x70;
         public const int ObjectiveCMessageSendFunctionShift = 4;
+        // Uses the same bit as IsObjectiveCMessageSendMask since we never have
```

Comment:
Can we assert that some how? In native code I would use the following, is there some mechanism in C#?

```
#if X86 && OBJC
#error "Not supported"
#endif
```

Reply:
I structured it in a way that `#elif` branches would cause it to fail to compile because of missing fields on the runtime side.

Reply:
https://github.com/dotnet/runtime/blob/b334da1c17b789b23a30b57abc242d48175b59f6/src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs#L646-L652